File: Models/DeckType.cs
using System.Collections.ObjectModel;

namespace scrum_poker_app.Models;

public class DeckType
{
    public int ID { get; set; }

    public string Name { get; set; } = "";
    
    public string Description { get; set; } = "";
    
    public ImageFileDimensions? Preview { get; set; }
    
    public Collection<CardDefinition>? Cards { get; set; }
    
    public Collection<OrderedSheetDefinition>? Sheets { get; set; }
}

File: ClientApp/src/app/deck-type.ts
import { CardDefinition } from './card-definition';
import { ImageFileDimensions } from './image-file-dimensions';
import { OrderedSheetDefinition } from './ordered-sheet-definition';

export interface DeckType {
  id: number;
  name: string;
  description: string;
  preview: ImageFileDimensions;
  cards: CardDefinition[];
  sheets: OrderedSheetDefinition[];
}
